pkg/graphql/execution_engine.go

```diff
@@ -129,6 +166,17 @@ func (e *ExecutionEngine) ExecuteWithWriter(ctx context.Context, operation *Requ
 	return poolExecutor.Execute(executionContext, plan, writer)
 }
 
+func (e *ExecutionEngine) operationID(operation *Request) (uint64, error) {
+	hash64 := e.hash64Pool.Get().(hash.Hash64)
+	printer := e.printerPool.Get().(*astprinter.Printer)
+	err := printer.Print(&operation.document, &e.schema.document, hash64)
```

Comment:
I was looking at the internals of the `printer.Print` to see if it leaves any kind of state, since we re-use the object, and found this piece of code:
```
...
if !p.registered {
  p.walker.SetVisitor(&p.visitor)
}
```
Once `registred` boolean value is set, it never gets released. I wonder if it can cause the issue, and maybe printer object should have a `Reset`  function, or similar.

Reply:
I think this is fine. SetVisitor simply ties this one printer visitor to the walker. They never get detached from each other. It's worth noticing that printer and walker is not thread safe which is intentional. We (re-) use it with a sync pool so that's fine then.